### PR TITLE
RES-863: Experiment with OpenAI and reinforcement learning

### PR DESCRIPTION
**Pull request by lscheinkman**
@subutai Please review
This PR depends on `LayerNorm` PR #967 so ignore the first 4 commits that are part of the other PR.

The relevant commit for this PR is b99adfe

### CODE REVIEW

File: projects/sparse_rl/sparse_dqn.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+                           useBatchNorm=True,
+                           useSoftmax=False)
+
+
+  def forward(self, x):
+    return self.model(x)
+
+
+
+class DQN(object):
```
Comment (subutai) [inline projects/sparse_rl/sparse_dqn.py]: This looks pretty cool, and general. Should we create a frameworks/rl and move the DQN class and support stuff there into separate files? Could potentially move the open AI utils there too.

Comment (lscheinkman) [reply projects/sparse_rl/sparse_dqn.py]: Done

Comment (subutai) [reply projects/sparse_rl/sparse_dqn.py]: Thanks!

**Comment by lscheinkman** [comment]
Isolated 'LayerNorm' change to #967 so we could merge this PR

**Pull request merged**